### PR TITLE
[AIRFLOW-4122] Remove chain function

### PR DESCRIPTION
**Pull request by zhongjiajie**
Bit operation like `>>` or `<<` are suggested
to set dependency, which visual and easier to
explain. and have multiple ways is confusion

Make sure you have checked _all_ steps below.

### Jira

- [x] My PR addresses the following [Airflow Jira](https://issues.apache.org/jira/browse/AIRFLOW/) issues and references them in the PR title. For example, "\[AIRFLOW-XXX\] My Airflow PR"
  - In case you are fixing a typo in the documentation you can prepend your commit with \[AIRFLOW-4122\], code changes always need a Jira issue.
  - In case you are proposing a fundamental code change, you need to create an Airflow Improvement Proposal ([AIP](https://cwiki.apache.org/confluence/display/AIRFLOW/Airflow+Improvements+Proposals)).
  - In case you are adding a dependency, check if the license complies with the [ASF 3rd Party License Policy](https://www.apache.org/legal/resolved.html#category-x).

### Description

- [x] Here are some details about my PR, including screenshots of any UI changes:

I had submit https://github.com/apache/airflow/pull/4779 . In PR comment, I was suggest close this PR and submit new one to remove function `chain` in `airflow.utils.heplers`, due to Airflower think use bit operation like `>>` or `<<` visual and easier to explain, and have multiple ways is confusion to users. BTW, this function was created 3 year ago and never get updated.

### Tests

- [x] My PR adds the following unit tests __OR__ does not need testing for this extremely good reason:

### Commits

- [x] My commits all reference Jira issues in their subject lines, and I have squashed multiple commits if they address the same issue. In addition, my commits follow the guidelines from "[How to write a good git commit message](http://chris.beams.io/posts/git-commit/)":
  1. Subject is separated from body by a blank line
  1. Subject is limited to 50 characters (not including Jira issue reference)
  1. Subject does not end with a period
  1. Subject uses the imperative mood ("add", not "adding")
  1. Body wraps at 72 characters
  1. Body explains "what" and "why", not "how"

### Documentation

- [x] In case of new functionality, my PR adds documentation that describes how to use it.
  - When adding new operators/hooks/sensors, the autoclass documentation generation needs to be added.
  - All the public functions and the classes in the PR contain docstrings that explain what it does
  - If you implement backwards incompatible changes, please leave a note in the [Updating.md](https://github.com/apache/airflow/blob/master/UPDATING.md) so we can assign it to a appropriate release

### Code Quality

- [x] Passes `flake8`

### CODE REVIEW

**Comment by zhongjiajie** [comment]
PTAL @Fokko @feng-tao @XD-DENG thanks, This PR remove `chain` function in `airlfow.utils.helpers` that we discuss in https://github.com/apache/airflow/pull/4779.

**Review by feng-tao — APPROVED** [review]
lgtm

**Comment by feng-tao** [comment]
will merge it tomorrow if there is no other comment. thanks @zhongjiajie

File: UPDATING.md
Comment (feng-tao) [inline UPDATING.md]: nit:  just called it `Deprecation chain function `

We don't know whether this change will be picked in the new release.

File: UPDATING.md
Comment (feng-tao) [inline UPDATING.md]: nit: how about
```
Bit operation like `>>` or `<<` are recommended for setting the dependency, which is easier to explain.
The `airflow.utlis.helpers.chain` function will be deprecated.
```
?

File: UPDATING.md
Comment (feng-tao) [reply UPDATING.md]: WDYT

File: UPDATING.md
Comment (feng-tao) [reply UPDATING.md]: WDYT

File: UPDATING.md
Comment (zhongjiajie) [reply UPDATING.md]: Agree with that.

File: UPDATING.md
Comment (zhongjiajie) [reply UPDATING.md]: I will change it. thanks.

**Comment by zhongjiajie** [comment]
@feng-tao I change `UPDATING.md` and skip CI, PTAL thanks.

**Comment by feng-tao** [comment]
LGTM

**Comment by zhongjiajie** [comment]
@feng-tao thanks.

File: UPDATING.md
Comment (ashb) [inline UPDATING.md:30]: FYI: A function is deprecated when it still exists and works but gives a warning. This is not what we did in this PR.

Comment (feng-tao) [reply UPDATING.md:30]: @ashb , just create a minor pr(https://github.com/apache/airflow/pull/4953) to correct. Thanks for pointing out. I should review more carefully.

Comment (zhongjiajie) [reply UPDATING.md:30]: @ash @feng-tao Sorry for the incorrect describe I add.

**Pull request merged**
